nibabel/spatialimages.py

```diff
@@ -867,6 +872,51 @@ def from_image(klass, img):
                      klass.header_class.from_header(img.header),
                      extra=img.extra.copy())
 
+    @classmethod
+    def is_valid_extension(klass, ext):
+        return np.any([ft[1] == ext.lower() for ft in klass.files_types])
+
+    @classmethod
+    def is_valid_filename(klass, filename):
+        froot, ext, trailing = splitext_addext(filename, klass._compressed_exts)
+        return klass.is_valid_extension(ext)
+
+    @classmethod
+    def is_image(klass, filename, sniff=None):
+        froot, ext, trailing = splitext_addext(filename, klass._compressed_exts)
+
+        if not klass.is_valid_extension(ext):
+            return False, sniff
+        elif (getattr(klass.header_class, 'sniff_size') is None or
+              getattr(klass.header_class, 'is_header') is None):
```

Comment:
Check if `None` rather than do a `hasattr`, so that properties can be either unset, or reset to `None` (which is easier when subclassing).